src/arkode/arkode_mristep.c

```diff
@@ -1551,50 +1868,82 @@ int mriStep_TakeStep(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
     case (MRISTAGE_DIRK_FAST):
       retval = mriStep_StageDIRKFast(ark_mem, step_mem, is, nflagPtr);
       break;
+    case (MRISTAGE_STIFF_ACC): retval = ARK_SUCCESS; break;
     }
     if (retval != ARK_SUCCESS) { return (retval); }
 
 #ifdef SUNDIALS_LOGGING_EXTRA_DEBUG
     SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
-                       "ARKODE::mriStep_TakeStep", "slow stage", "z_%i(:) =", is);
+                       "ARKODE::mriStep_TakeStepMRIGARK", "slow stage",
+                       "z[%i] =", is);
     N_VPrintFile(ark_mem->ycur, ARK_LOGGER->debug_fp);
 #endif
 
     /* apply user-supplied stage postprocessing function (if supplied) */
-    if (ark_mem->ProcessStage != NULL)
+    if ((ark_mem->ProcessStage != NULL) &&
+        (step_mem->stagetypes[is] != MRISTAGE_STIFF_ACC))
     {
-      retval = ark_mem->ProcessStage(ark_mem->tcur, ark_mem->ycur,
-                                     ark_mem->user_data);
+      retval = ark_mem->ProcessStage(tf, ark_mem->ycur, ark_mem->user_data);
       if (retval != 0) { return (ARK_POSTPROCESS_STAGE_FAIL); }
     }
 
     /* conditionally reset the inner integrator with the modified stage solution */
-    if ((step_mem->stagetypes[is] != MRISTAGE_ERK_FAST) ||
-        (ark_mem->ProcessStage != NULL))
+    if (step_mem->stagetypes[is] != MRISTAGE_STIFF_ACC)
     {
-      retval = mriStepInnerStepper_Reset(step_mem->stepper, ark_mem->tcur,
-                                         ark_mem->ycur);
-      if (retval != ARK_SUCCESS) { return (ARK_INNERSTEP_FAIL); }
+      if ((step_mem->stagetypes[is] != MRISTAGE_ERK_FAST) ||
+          (ark_mem->ProcessStage != NULL))
+      {
+        retval = mriStepInnerStepper_Reset(step_mem->stepper, tf, ark_mem->ycur);
+        if (retval != ARK_SUCCESS)
+        {
+          arkProcessError(ark_mem, ARK_INNERSTEP_FAIL, __LINE__, __func__,
+                          __FILE__, "Unable to reset the inner stepper");
+          return (ARK_INNERSTEP_FAIL);
+        }
+      }
     }
 
-    /* Compute updated slow RHS except at last stage which is the new solution.
-     * The new solution RHS evaluation happens in arkCompleteStep */
-    if (is < step_mem->stages - 1 && step_mem->stage_map[is] > -1)
+    /* Compute updated slow RHS, except:
+       1. at last stage which is the new solution, since that RHS occurs in arkCompleteStep.
+       2. if the stage is excluded from stage_map
+       3. if the next stage has "STIFF_ACC" type, and temporal estimation is disabled */
```

Comment:
If you look back in [Sandu's MRI-GARK paper](https://doi.org/10.1137/18M1205492), there are multiple tables (e.g., MRI-GARK-ESDIRK34a) where the Butcher table is stiffly accurate, and so when he converts this to the $\Gamma$ coefficients it has a zero row, and $\Delta c = 0$.  In our previous implementation, we just trimmed off that stage when entering the coefficient tables and called it a day.  However, to support embeddings I needed that row to be added back in, but due to the structure the corresponding stage solution would just be a copy of the previous stage.  I added the `MRISTAGE_STIFF_ACC` type so that we could:

1. insert the tables to match those in the publication,
2. correctly compute the embeddings,
3. eliminate unnecessary stage vector storage,
4. eliminate unnecessary computation.

There might have been a different set of logic statements to ensure 2-4, but the current approach does it all.